source/extensions/access_loggers/grpc/grpc_access_log_utils.cc

```diff
@@ -235,6 +236,23 @@ void Utility::extractCommonAccessLogProperties(
   if (stream_info.dynamicMetadata().filter_metadata_size() > 0) {
     common_access_log.mutable_metadata()->MergeFrom(stream_info.dynamicMetadata());
   }
+
+  for (const auto& key : config.filter_state_objects_to_log()) {
+    if (stream_info.filterState().hasDataWithName(key)) {
+      const auto& obj =
+          stream_info.filterState().getDataReadOnly<StreamInfo::FilterState::Object>(key);
+      ProtobufTypes::MessagePtr serialized_proto = obj.serializeAsProto();
+      if (serialized_proto != nullptr) {
+        auto& filter_state_objects = *common_access_log.mutable_filter_state_objects();
+        ProtobufWkt::Any& any = filter_state_objects[key];
+        if (dynamic_cast<ProtobufWkt::Any*>(serialized_proto.get()) != nullptr) {
+          any.Swap(dynamic_cast<ProtobufWkt::Any*>(serialized_proto.get()));
+        } else {
+          any.PackFrom(*serialized_proto);
```

Comment:
Shouldn't any packfrom any override both type and value bytes? Why special case?

Reply:
If the returned MessagePtr is already an `Any`, use it directly. This is helpful for WASM extensions that has protos unknown to Envoy and also avoid having `Any` in `Any`.

Reply:
OK, if it avoids any.PackFrom(any) nesting them (why does the library do that :/)